File: 1_file_types/parse_files.js
const fs = require('fs');
const Papa = require('papaparse');
const path = require('path');
const { XMLParser } = require('fast-xml-parser');
const yaml = require('js-yaml')

const parseCSV = async () => {
    const p = path.resolve('me.csv');
    const file = fs.readFileSync(p, 'utf-8');
    
    const res = Papa.parse(file, {
        dynamicTyping: true,
        header: true
    });
    return res.data;
};

const parseJSON = () => {
    const p = path.resolve(__dirname, 'me.json');
    const file = fs.readFileSync(p);
    const parsedJSON = JSON.parse(file);
    return parsedJSON;
};

const parseXML = () => {
    const p = path.resolve(__dirname, 'me.xml');
    const file = fs.readFileSync(p, { encoding: 'utf-8' });
    const parser = new XMLParser();
    const parsedXML = parser.parse(file);
    return parsedXML;
};

const parseTXT = () => {
    const p = path.resolve(__dirname, 'me.txt');
    const file = fs.readFileSync(p, { encoding: 'utf-8' });
    
    let obj = {};
    const lines = file.split('\n');
    lines.forEach(line => {
        const _line = line.trim().split('=');
        if (_line[1].split(',').length > 1) {
            obj[_line[0]] = _line[1].split(',');
        } else {
            obj[_line[0]] = _line[1];
        }
    });
    return obj;
};

const parseYAML = () => {
    const p = path.resolve(__dirname, 'me.yaml');
    const parsedYAML = yaml.load(fs.readFileSync(p, 'utf-8'));
    return parsedYAML;
};
console.log(__dirname)
console.log(parseTXT())

File: data-format-translation/parse_files.py
import yaml, csv, json, xmltodict
import os

yaml_file = os.path.join(os.getcwd(),'me.yaml')
csv_file = os.path.join(os.getcwd(),'me.csv')
json_file = os.path.join(os.getcwd(),'me.json')
xml_file = os.path.join(os.getcwd(),'me.xml')
txt_file = os.path.join(os.getcwd(),'me.txt')

def parse_yaml(file_path=yaml_file):
    with open(file=file_path, mode='r') as f:
        yaml_obj = yaml.load(f, yaml.Loader)
    return yaml_obj

def parse_csv(file_path=csv_file):
    parsed_csv = []
    with open(file=file_path, mode='r') as f:
        parsed_csv = [ row for row in csv.DictReader(f) ]
    return parsed_csv
            
def parse_json(file_path=json_file):
    with open(file=file_path, mode='r') as f:
        parsed_json = json.load(f)
        return parsed_json

def parse_xml(file_path=xml_file):
    with open(file_path, mode='r') as f:
        file = f.read()
        parsed_xml = xmltodict.parse(file)
        return parsed_xml

def parse_txt(file_path=txt_file):
    parsed_txt = {}
    with open(file_path, mode='r') as f:
        rows = [ r.strip() for r in f.readlines() ]
        for row in rows:
            row = row.split('=')
            if len(row[1].split(',')) > 1:
                parsed_txt[row[0]] = [ i for i in row[1].split(',') ]
            else:
                parsed_txt[row[0]] = row[1]
    return parsed_txt